docs/pages/database-access/rbac.mdx

```diff
@@ -0,0 +1,134 @@
+    db_users: ["viewer", "editor"]
+
+    # Database names this role will be able to connect to.
+    #
+    # Note, this is not the same as the "name" field in "db_service", this is
+    # the database names within a particular database instance.
+    #
+    # Also note, this setting has effect only for PostgreSQL. It does not
+    # currently have any effect on MySQL databases/schemas.
+    db_names: ["main", "metrics", "postgres"]
+```
+
+It is possible to use wildcards to match any database names/users.
+
+For example, the following role permits access to any database/user within a
+production database except for the internal "postgres" database/user:
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: developer
+spec:
+  allow:
+    db_labels:
+      environment: ["prod"]
+    db_users: ["*"]
+    db_names: ["*"]
+  deny:
+    db_users: ["postgres"]
+    db_names: ["postgres"]
+```
+
+<Admonition type="note" title="Deny Rules">
+Deny rules will match greedily. In the example above, a database connection
+attempting to use "postgres" database account (regardless of database instance
+or database name) or "postgres" database name (regardless of database instance
+or database account) will be rejected.
+</Admonition>
+
+## Database Names
+
+There's a distinction in how different database servers handle logical databases
+which leads to a difference in how `db_names` role field is applied to a connection
+attempt.
+
+PostgreSQL supports multiple logical databases, and each logical database can
+contain multiple schemas. In order to change to a different database, a user
+disconnects from the current one and establishes a new connection. During a
+PostgreSQL connection attempt, `db_names` field is checked against the name
+of the logical database that the user is connecting to.
+
+In MySQL a logical "database" and a "schema" are synonyms for each other, and
+the scope of permissions a user has once connected is determined by the permission
+grants set on the account within the database. As such, `db_names` role field
+is not currently enforced on MySQL connection attempts.
+
+## Template Variables
+
+Similar to other role fields, `db_*` fields support templating variables.
+
+The `{{internal.db_users}}` and `{{internal.db_names}}` variables allow sharing
+allowed database accounts and names with remote clusters. They will be replaced
+with the respective properties of a remote user connecting from a root cluster.
+
+For example, suppose a user in the root cluster has the following role:
+
+```yaml
+spec:
+  allow:
+    db_users: ["postgres"]
+    db_names: ["postgres"]
+```
+
+The role on the leaf cluster can be set up to use the user's allowed database
+accounts and names:
+
+```yaml
+spec:
+  allow:
+    db_users: ["{% raw %}{{internal.db_users}}{% endraw %}"]
+    db_names: ["{% raw %}{{internal.db_names}}{% endraw %}"]
+```
+
+The `{{external.xyz}}` variables are replaced with values from external [SSO](../enterprise/sso/ssh-sso.mdx)
```

Comment:
I feel external variable examples should go above internal/trusted cluster examples. Typically you would configure your root cluster first, then trusted clusters.